lib/emscripten/src/lib.rs

```diff
@@ -351,6 +351,8 @@ pub fn run_emscripten_instance(
     // println!("running emscripten instance");
 
     if let Some(ep) = entrypoint {
+        // check if entrypoint exist
+        instance.dyn_func(&ep)?;
```

Comment:
Is it possible to use the result of this to make the call below?  That way it doesn't  need a comment and is more clear in my opinion.

Reply:
the result of this function is not useful in that case but i can replace by `let _func = instance.dyn_func(&ep)?;` if it's more clear

Reply:
No, giving it a name without using it doesn't help.

Hmm, perhaps I'm missing something but it looks like this should be possible unless you're saying that there's a borrowing issue.

```Rust
let arg = unsafe { allocate_cstr_on_stack(instance.context_mut(), args[0]).0 };
let func = instance.dyn_func(&ep)?;
debug!("Running entry point: {}", &ep);
func.call(&[Value::I32(arg as i32)])?;
```

Isn't that the same as the existing code?  The types line up.  The only issue I can see is a borrow checking one, we can fix that by putting the `arg` statement first.  That's a bit more guest-memory-leaky than the current code, though the current code leaks guest-memory anyways and it only does that if the function doesn't exist.

If we can't use the result then it seems like using `resolve_func` instead of `dyn_func` makes more sense.  `dyn_func` does extra work to return something we can call, no point doing that if we aren't going to call it.

Reply:
There is actually 2 issue in this code minimum, first one is because we are not checking is the function exist. Second one is because we try to retrieve arguments that doesn't exist so we need to do some checks before 
``` rust
let arg = unsafe { allocate_cstr_on_stack(instance.context_mut(), args[0]).
```

I'll work on that today.
Thanks

Reply:
Please check the new version ;)